docs: vite experimental docs (#1191)
seek-oss/sku: docs/docs/using-vite.md

import { LoadableProvider, preloadAll } from 'sku/vite/loadable';

import App from './App';

export default {
  render: async ({ options, renderContext, site, url }) => {
    const { loadableCollector } = renderContext;

    await preloadAll();

    return renderToPipeableStream(
      <LoadableProvider value={loadableCollector}>
        <StaticRouter location={url} context={{}}>
          <App site={site} />
        </StaticRouter>
      </LoadableProvider>,
      options,
    );
  },

  provideClientContext: ({ site }) => ({
    site,
  }),
} satisfies ViteRender;
```

### Code splitting

Code splitting is possible with the new `sku/vite/loadable` entrypoint.

The new `sku/vite/loadable` entrypoint relies on React's [`<Suspense />`](https://react.dev/reference/react/Suspense) component for the loading of a fallback state.

**Example of using `sku/vite/loadable`**

```tsx
import { Suspense } from 'react';
import { loadable } from 'sku/vite/loadable';

const Home = loadable(() => import('./Home'));

export default () => (
  <div>
    <Suspense fallback={<div>Loading Home...</div>}>
      <Home />
    </Suspense>
  </div>
);
```

### Using vite types

> If `vite/client` reference types are needed in your project, you can import them from `sku/vite/client`.

Reviewer: ```suggestion
If [`vite/client` types](https://vite.dev/guide/features#client-types) are needed in your project, you can reference them via the `sku/vite/client` entrypoint:
```

Should we also be more specific about where to put this `<reference />`? Maybe `client.tsx`?
